checks/dangerous_workflow.go

```diff
@@ -16,17 +16,47 @@ package checks
 
 import (
 	"fmt"
+	"regexp"
 	"strings"
 
 	"github.com/rhysd/actionlint"
 
 	"github.com/ossf/scorecard/v3/checker"
 	"github.com/ossf/scorecard/v3/checks/fileparser"
+	sce "github.com/ossf/scorecard/v3/errors"
 )
 
 // CheckDangerousWorkflow is the exported name for Dangerous-Workflow check.
 const CheckDangerousWorkflow = "Dangerous-Workflow"
 
+func containsUntrustedContextPattern(variable string) bool {
```

Comment:
can you add as comment the link where you took those from? Either the security lab blog pst of the GitHub doc would do, I think.